### PR TITLE
Run test on freebsd

### PR DESCRIPTION
**Pull request by yuichiro-naito**
I fixed to run all tests have been passed on FreeBSD 11-R.
I think it is useful for future development of ruby-build.
Will you merge this PR ?

### CODE REVIEW

File: script/brew-publish [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
```
Comment (mislav) [inline script/brew-publish]: Where is bash located on freebsd?

File: test/build.bats [hunk 2]
```diff
@@ -355,6 +367,7 @@ apply -p1 -i /my/patch.diff
 exec ./configure "\$@"
 CONF
 
+  stub uname '-s : echo Linux'
```
Comment (mislav) [inline test/build.bats]: What happens if `uname` is not stubbed?

File: test/test_helper.bash [hunk 3]
```diff
@@ -3,7 +3,7 @@ export TMP="$BATS_TEST_DIRNAME/tmp"
 if [ "$FIXTURE_ROOT" != "$BATS_TEST_DIRNAME/fixtures" ]; then
   export FIXTURE_ROOT="$BATS_TEST_DIRNAME/fixtures"
   export INSTALL_ROOT="$TMP/install"
-  PATH=/usr/bin:/usr/sbin:/bin/:/sbin
+  PATH="/usr/bin:/usr/sbin:/bin/:/sbin:$PATH"
```
Comment (mislav) [inline test/test_helper.bash]: Which important directory was missing from this list? I specifically didn't want to include the whole `$PATH`.

File: script/brew-publish (hunk 1 above)
Comment (yuichiro-naito) [inline script/brew-publish]: bash is installed in `/usr/local/bin/bash`.
It's an optional package and not included in FreeBSD base system.

File: test/build.bats (hunk 2 above)
Comment (yuichiro-naito) [inline test/build.bats]: When `uname -s` returns `FreeBSD`,
ruby-build adds configure option '--with-readline-dir=/usr/local'.
It is a result of my previous Pull Request.

Some error message are shown like below.

```
✗ yaml is installed for ruby
   (from function `assert_equal' in file test/test_helper.bash, line 139,
    from function `assert_output' in file test/test_helper.bash, line 148,
    from function `assert_build_log' in file test/build.bats, line 58,
    in test file test/build.bats, line 76)
     `assert_build_log <<OUT' failed
   expected: yaml-0.1.6: --prefix=${TMP}/install
   make -j 2
   make install
   ruby-2.0.0: --prefix=${TMP}/install
   make -j 2
   make install
   actual:   yaml-0.1.6: --prefix=${TMP}/install
   make -j 2
   make install
   ruby-2.0.0: --prefix=${TMP}/install --with-readline-dir=/usr/local
   make -j 2
   make install
```

File: test/test_helper.bash (hunk 3 above)
Comment (yuichiro-naito) [inline test/test_helper.bash]: `/usr/local/bin` is the most important.

I'm guessing some people might compile and install bash
to other directory like `/opt/pkg/bin`.
FreeBSD ports tree supports $PREFIX definition.

c.f. https://www.freebsd.org/doc/en/books/porters-handbook/porting-prefix.html

I don't use this option,
and if you don't want to include whole $PATH,
just adding `/usr/local/bin` satisfies me.

Comment (mislav) [inline test/test_helper.bash]: You you change the list to:

```
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

**Comment by yuichiro-naito** [comment]
Thanks for the comment.
I fixed the code.

**Comment by mislav** [comment]
Thank you!

**Pull request merged**